generated/proto/index/index.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+package index;
+
+message IndexInfo {
+  message SegmentFileInfo {
```

Comment:
nit: could you avoid the nested type declarations. would make this a little cleaner to follow

Reply:
Sure thing.